Lib/test/test_unicode.py

```diff
@@ -2819,6 +2819,16 @@ def test_asucs4(self):
             self.assertEqual(unicode_asucs4(s, len(s), 1), s+'\0')
             self.assertEqual(unicode_asucs4(s, len(s), 0), s+'\uffff')
 
+    # Test PyUnicode_AsUTF8()
+    @support.cpython_only
+    def test_asutf8(self):
+        from _testcapi import unicode_asutf8
+
+        self.assertEqual(unicode_asutf8('abc'), 'abc')
```

Comment:
The function encodes to UTF-8, so the result should be a bytes string, not Unicode. Use PyString_FromString() rather than PyUnicode_FromString() in _testcapi.

Reply:
The others similar C API tests also return unicodes other than bytes which in some sense also tests the counter part decoding function, though I don't very understand why designed like this.

Reply:
Either fix other tests, or leave them unchanged. But I would prefer to not add new buggy tests :-)

Reply:
So returning the decoded bytes string is right way? I am not sure I understand it clearly.

Reply:
Return bytes, not unicodes. Test decode functionality separately.